### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

**Comment by dantegd** [comment]
rerun tests

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Java:
+ - 'java/**'
+
+Ops:
+ - '.github/**'
+ - 'ci/**'
+ - 'conda/**'
+ - '**/Dockerfile'
+ - '**/.dockerignore'
+ - 'docker/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:25]: This section should be removed

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+libraft:
+ - 'cpp/**'
+
+CMake:
+ - '**/CMakeLists.txt'
+ - '**/cmake/**'
+
+Java:
+ - 'java/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:17]: These labels should match what labels are in this repo and the setup of the `.github/CODEOWNERS` for this repo

**Review by mike-wendt — CHANGES_REQUESTED** [review]

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
This repo doesn't have any of the labels listed in `labeler.yml`. I'd check with Mike to see how to handle missing labels. Some can likely be added, but a few may need to be modified (i.e. `ci` -> `gpuCI`).

**Comment by dantegd** [comment]
@ajschmidt8 creating the missing labels (and and any modifications) here to match cuml/cugraph would be perfect in my opinion. They don't exist yet mainly due to not having been needed often in issues/PRs (for example gpuCI), but having them would be a net positive.

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
@ajschmidt8 can you re-review?

**Comment by mike-wendt** [comment]
for reference here are the equivalent cuml/cugraph PRs:
- rapidsai/cuml#3407
- rapidsai/cugraph#1358

@dantegd @rlratzel can you chime in on if there need to be any changes and review when you can? thanks!

**Review by ajschmidt8 — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]
LGTM from cugraph's perspective.

@dantegd we've also added a `docs` label for `'**/*.md'`patterns, but maybe raft doesn't need that yet?

**Comment by mike-wendt** [comment]
@gpucibot merge

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Pull request merged**